src/rules/custom-property-empty-line-before/__tests__/index.js

```diff
@@ -0,0 +1,147 @@
+  ruleName,
+  config: ["always"],
+
+  accept: [ {
+    code: "a {\n\n --custom-prop: value;\n}",
+  }, {
+    code: "a {\r\n\r\n --custom-prop: value;\r\n}",
+  } ],
+
+  reject: [ {
+    code: "a {\n --custom-prop: value;\n}",
+    message: messages.expected,
+  }, {
+    code: "a {\r\n --custom-prop: value;\r\n}",
+    message: messages.expected,
+  } ],
+
+}))
+
+testRule(rule, {
+  ruleName,
+  config: [ "always", { except: ["first-nested"] } ],
+
+  accept: [ {
+    code: "a { --custom-prop: value;\n}",
+  }, {
+    code: "a {\n --custom-prop: value;\n}",
+  }, {
+    code: "a {\r\n --custom-prop: value;\r\n}",
+  } ],
+
+  reject: [ {
+    code: "a {\n\n --custom-prop: value;\n}",
+    message: messages.rejected,
+  }, {
+    code: "a {\r\n\r\n --custom-prop: value;\r\n}",
+    message: messages.rejected,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [ "always", { except: ["after-comment"] } ],
+
+  accept: [ {
+    code: "a {\n\n --custom-prop: value;\n}",
+  }, {
+    code: "a {/* I am a comment */ \n --custom-prop2: value;}",
+  }, {
+    code: "a {/* I am a comment */ \r\n --custom-prop2: value;}",
+  } ],
+
+  reject: [ {
+    code: "a {\n\n --custom-prop: value;\n /* I am a comment */ \n\n --custom-prop2: value;}",
+    message: messages.rejected,
+  }, {
+    code: "a {\n /* I am a comment */ \r\n\r\n --custom-prop2: value;}",
+    message: messages.rejected,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [ "always", { except: ["after-custom-property"] } ],
+
+  accept: [ {
+    code: "a {\n\n --custom-prop: value;\n}",
+  }, {
+    code: "a {\n\n --custom-prop:value; \n --custom-prop2: value;}",
+  }, {
+    code: "a {\n\n --custom-prop:value; \r\n --custom-prop2: value;}",
+  } ],
+
+  reject: [ {
+    code: "a {\n\n --custom-prop:value;\n\n --custom-prop2: value;}",
+    message: messages.rejected,
+  }, {
+    code: "a {\n\n --custom-prop: value;\r\n\r\n --custom-prop2: value;}",
+    message: messages.rejected,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [ "always", {
+    except: [
+      "first-nested",
+      "after-comment",
+      "after-custom-property",
+    ] } ],
+
+  accept: [{ code: "a {\n --custom-prop: value; \n --custom-prop2: value; \n /* comment */ \n --custom-prop3: value;\n\n @extends 'x';\n\n --custom-prop4: value; \n & b {\n prop: value;\n } \n\n --custom-prop5: value; \n }" }],
+
+})
+
+testRule(rule, {
+  ruleName,
+  config: ["never"],
+
+  accept: [ {
+    code: "a {\n --custom-prop: value;\n}",
+  }, {
+    code: "a {\r\n --custom-prop: value;\r\n}",
+  }, {
+    code: "a {\n --custom-prop: value; \n --custom-prop2: value; \n\n /* comment */ \n --custom-prop3: value; \n\n @extends 'x'; \n --custom-prop4: value; \n\n & b {\n prop: value; \n } \n --custom-prop5: value;\n }",
+  } ],
+
+  reject: [ {
+    code: "a {\n\n --custom-prop: value;\n}",
+    message: messages.rejected,
+  }, {
+    code: "a {\r\n\r\n --custom-prop: value;\r\n}",
```

Comment:
Can we get an `accept` and `reject` test for a custom property that follows:

1. A declaration e.g. `top: 10px;`
2. A nested at-rule. `@extends .class;`
3. A dollar-variable. `$var: value`